workshop/content/exercises/04-remediations.md

```diff
@@ -1,9 +1,391 @@
+an explanation on the `spec` section, you can do `oc explain complianceremediation.spec`.
+
+To list al the remediations that were generated as part of a specific suite run, you can do:
+
+```
+oc get complianceremediations -l complianceoperator.openshift.io/suite=<suite name>
+```
+
+Let's see the remediations that came from the scan we did in the previous exercise
+
+```
+$ oc get complianceremediations -l compliance.openshift.io/suite=periodic-e8
+NAME                                                       STATE
+rhcos4-e8-master-audit-rules-dac-modification-chmod        NotApplied
+rhcos4-e8-master-audit-rules-dac-modification-chown        NotApplied
+rhcos4-e8-master-audit-rules-execution-chcon               NotApplied
+rhcos4-e8-master-audit-rules-execution-restorecon          NotApplied
+rhcos4-e8-master-audit-rules-execution-semanage            NotApplied
+rhcos4-e8-master-audit-rules-execution-setfiles            NotApplied
+...
+```
+
+You'll notice that none of the remediations are applied. This is the default behavior, as
+we don't want the operator unexpectedly changing the cluster's configuration.
+
+Let's inspect one of the generated remediations and see what it does:
+
+```
+$ oc get complianceremediation rhcos4-e8-worker-sysctl-kernel-dmesg-restrict -o yaml
+apiVersion: compliance.openshift.io/v1alpha1
+kind: ComplianceRemediation
+metadata:
+  creationTimestamp: "2020-09-08T11:47:09Z"
+  generation: 1
+  labels:
+    compliance.openshift.io/scan-name: rhcos4-e8-worker
+    compliance.openshift.io/suite: periodic-e8
+    machineconfiguration.openshift.io/role: worker
+  name: rhcos4-e8-worker-sysctl-kernel-dmesg-restrict
+  namespace: openshift-compliance
+  ownerReferences:
+  - apiVersion: compliance.openshift.io/v1alpha1
+    blockOwnerDeletion: true
+    controller: true
+    kind: ComplianceCheckResult
+    name: rhcos4-e8-worker-sysctl-kernel-dmesg-restrict
+    uid: 83aed9ae-2f08-4a30-a850-368385d1d00a
+  resourceVersion: "28256"
+  selfLink: /apis/compliance.openshift.io/v1alpha1/namespaces/openshift-compliance/complianceremediations/rhcos4-e8-worker-sysctl-kernel-dmesg-restrict
+  uid: cb8e972a-4257-47fb-817d-db60b1fd20bc
+spec:
+  apply: false
+  current:
+    object:
+      apiVersion: machineconfiguration.openshift.io/v1
+      kind: MachineConfig
+      spec:
+        config:
+          ignition:
+            version: 2.2.0
+          storage:
+            files:
+            - contents:
+                source: data:,kernel.dmesg_restrict%3D1
+              filesystem: root
+              mode: 420
+              path: /etc/sysctl.d/75-sysctl_kernel_dmesg_restrict.conf
+  outdated: {}
+status:
+  applicationState: NotApplied
+```
+
+You'll notice that this remediation merely creates a MachineConfig object that sets the `dmesg_restrict` 
+sysctl. You'll also notice that there is a controller set in the `ownerReferences` key in the metadata
```

Comment:
Is the owner object commonly called a controller?

Reply:
not always. The `controller` field mainly indicates that there is an object whose controller owns the object. We seem to be setting this value, hence why I mentioned it.